### PR TITLE
Create Uni V2 Withdraw Hook

### PR DESCRIPTION
**Pull request by JeanNeiverth**
- Change functionality to withdraw uniswap v2 instead of balancer v2
- Support arbitrum and mainnet
- Didn't change the name of the app folder to make it more reviewable

I'm using the WETH/USDC pair on Arbitrum for tests:
https://arbiscan.io/address/0xf64dfe17c8b87f012fcf50fbda1d62bfa148366a#readContract

Still not tested in mainnet

### CODE REVIEW

File: apps/withdraw-balancer-v2/src/utils/getUserLpMints.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import { SupportedChainId } from "@cowprotocol/cow-sdk";
+import { gql } from "urql";
+import { thegraphClientsMap } from "./thegraphClientsMap";
+
+export interface UserLpMintsReturn {
+  mints: {
+    sender: string;
+    to: string;
+    pair: {
```
Comment (yvesfracari) [inline apps/withdraw-balancer-v2/src/utils/getUserLpMints.ts:9]: I am not familiar with this subgraph but make sure it handles the transfer case that I mentioned before.

File: apps/withdraw-balancer-v2/src/utils/readUserBalances.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    return balances.map((result) =>
+      result.status === "success"
+        ? BigNumber.from(result.result)
+        : BigNumber.from("0"),
+    );
+  } catch (error) {
+    console.error("Error reading balances:", error);
+    return new Array(contracts.length).fill(BigInt(0));
+  }
+}
```
Comment (yvesfracari) [inline apps/withdraw-balancer-v2/src/utils/readUserBalances.ts:32]: I think that you can use same function that we used to read the other token information. 

Also, this function doesn't takes the balancesDiif (due previous hooks) into account.

File: apps/withdraw-balancer-v2/src/utils/thegraphClientsMap.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import { cacheExchange, fetchExchange } from "@urql/core";
+import { createClient } from "urql";
+
+const arbitrumClient = createClient({
+  url: `https://gateway.thegraph.com/api/${process.env.NEXT_PUBLIC_THEGRAPH_API_KEY}/subgraphs/id/CStW6CSQbHoXsgKuVCrk3uShGA4JX3CAzzv2x9zaGf8w`,
+  exchanges: [cacheExchange, fetchExchange],
+});
+
+const mainnetClient = createClient({
+  url: `https://gateway.thegraph.com/api/${process.env.NEXT_PUBLIC_THEGRAPH_API_KEY}/subgraphs/id/A3Np3RQbaBA6oKJgiwDJeo5T3zrYfGHPWFYayMwtNDum`,
```
Comment (yvesfracari) [inline apps/withdraw-balancer-v2/src/utils/thegraphClientsMap.ts:11]: This API would be accessible on the FE side, is this an issue?

File: packages/cow-hooks-ui/src/withdraw/PoolBalancePreview.tsx [hunk 4]
```diff
@@ -85,7 +85,7 @@ export function PoolBalancesPreview({
                         balance.token.decimals,
                       ),
                     )}
-                    fiatValue={balance.fiatAmount}
+                    fiatValue={withdrawBalance[index].fiatAmount}
```
Comment (yvesfracari) [inline packages/cow-hooks-ui/src/withdraw/PoolBalancePreview.tsx:88]: Is this another bug fix? 🤔

**Review by yvesfracari — COMMENTED** [review]
Thanks, good work! Just let a few comments

File: packages/cow-hooks-ui/src/withdraw/PoolBalancePreview.tsx (hunk 4 above)
Comment (JeanNeiverth) [reply packages/cow-hooks-ui/src/withdraw/PoolBalancePreview.tsx:88]: yep, before this modification, changing the withdraw percentage was not updating the displayed fiat amount

File: apps/withdraw-balancer-v2/src/utils/readUserBalances.ts (hunk 2 above)
Comment (JeanNeiverth) [reply apps/withdraw-balancer-v2/src/utils/readUserBalances.ts]: With the last update, I refactored it to read multiple info from many contracts, so It got a little more complex.

About the balanceDiff, I have my doubts if would be worth to consider it... I mean, it seems the chances are pretty small of someone's LP tokens change before a pre-hook, isn't it?

File: apps/withdraw-balancer-v2/src/utils/readPairsData.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  reserve1: BigNumber;
+  totalSupply: BigNumber;
+}
+
+export async function readPairsData(
+  userAddress: string,
+  pairAddresses: string[],
+  client: PublicClient,
+): Promise<PairData[]> {
+  try {
```
Comment (yvesfracari) [inline apps/withdraw-balancer-v2/src/utils/readPairsData.ts:20]: Make sure you consider balancesDiff from the frame context when you use this user balance.

File: apps/withdraw-balancer-v2/src/utils/readTokensData.ts
Comment (yvesfracari) [inline apps/withdraw-balancer-v2/src/utils/readTokensData.ts:1]: Same here:

> Make sure you consider balancesDiff from the frame context when you use this user balance.

**Review by yvesfracari — COMMENTED** [review]
Thanks @JeanNeiverth . Posted a few comments, let me know if you have any questions.

**Pull request merged**